### DISCUSSION TITLE
The relationship between viewId and viewState

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#3501] [Q&A] [answered]
Are these two properties related? If so, how to convert to each other?

**Comment by pmconne** [comment, marked as answer]
Can you be more specific please? What class/object are you talking about?

**Reply by Snailya** [reply]
I want to use IModelViewportComponent as the content view. This component need to pass in viewState as parameter.
This would be done by calling UiFramework.setDefaultViewState at upper level, then call getDefaultViewState.
When doing this, I find that there's also a method called setDefaultViewId. So I'm curious about whether viewState has a one to one relationship with viewId.

**Reply by pmconne** [reply]
Thanks. Sadly those APIs appear to entirely lack documentation. My expectation is that the default view Id refers to the Id of a ViewDefinition element stored in the iModel, from which a ViewState can be obtained instead of the application supplying its own ViewState. @NancyMcCallB @bsteinbk.

**Reply by Snailya** [reply]
If so, could you provider some sample code to explain how to obtain ViewState from View Id?

**Reply by pmconne** [reply]
https://www.itwinjs.org/learning/frontend/views/#loading-views-from-an-imodel
